src/common/snippets/src/pass/mha_tokenization.cpp

```diff
@@ -0,0 +1,390 @@
+    // TODO: Add Reshape, FQ support
+    while (is_supported_op(interm_op)) {
+        // All supported intermediate ops have only one output port
+        // To verify output element type is enough because all supported intermediate ops have the same output element type as input type
+        if (interm_op->get_output_target_inputs(0).size() != 1 || !is_supported_tensor(interm_op->get_output_tensor(0)))
+            return false;
+
+        // Check for supported Broadcast op
+        if (interm_op->get_input_size() > 1) {
+            tokenize_broadcast(interm_op, ordered_ops);
+        }
+
+        ordered_ops.push_back(interm_op);
+        interm_op = interm_op->get_output_target_inputs(0).begin()->get_node()->shared_from_this();
+    }
+    return true;
+};
+
+ngraph::snippets::pass::TokenizeMHASnippets::TokenizeMHASnippets() {
+    MATCHER_SCOPE(TokenizeMHASnippets);
+
+    auto m_matmul0 = std::make_shared<ngraph::opset1::MatMul>(ngraph::pattern::any_input(ngraph::pattern::has_static_shape()),
+                                                              ngraph::pattern::any_input(ngraph::pattern::has_static_shape()));
+
+    register_matcher(std::make_shared<ngraph::pattern::Matcher>(m_matmul0, matcher_name),
+        [=](ngraph::pattern::Matcher &m) {
+        OV_ITT_SCOPED_TASK(ngraph::pass::itt::domains::SnippetsTransform, "Snippets::op::TokenizeMHASnippets")
+        auto& pattern_to_output = m.get_pattern_value_map();
+
+        // After some transformations, a different number of Constants for some operations may be created
+        // than the actual number of Constants during tokenization.
+        // To avoid unsupported number of non-scalar Constants in the future (plugin specific limitation)
+        // we should calculate potential number of non-scalar Constants that will be moved up from body.
+        // TODO: Need update this variable when FQ will be supported
+        size_t hidden_virtual_ports_count = 0;
+        // Default value is True because MHA pattern always requires Buffer op
+        bool need_buffer = true;
+        std::string fused_names;
+        ngraph::NodeVector ordered_ops;
+
+        /* ======== Matcher Pass ========== */
+
+        /****** Skeleton ******/
+        /* Skeleton on MHA-pattern is:
+         *              \     /
+         *              MatMul0
+         *                 |
+         *    Eltwise/Select/Reshape/FakeQuantize
+         *                 |
+         *              Softmax
+         *                 |
+         *    Eltwise/Select/Reshape/FakeQuantize
+         *                  \      /
+         *                   MatMul1
+         */
+        const auto matmul0 = ngraph::as_type_ptr<ngraph::opset1::MatMul>(pattern_to_output.at(m_matmul0).get_node_shared_ptr());
+        if (!matmul0 || matmul0->get_output_target_inputs(0).size() != 1 || matmul0->get_transpose_a() ||
```

Comment:
I discussed offline it with @jane-intel and we decided that for our case the current solution is better because when we use `Label`, a pass matches on each node and check for conditions. When we use explicit node for matching, pass matches only on this node.

Reply:
Makes sense, thanks a lot!